Add EIP: Financial Bonds (#7092)
ethereum/EIPs: EIPS/eip-7092.md

---
eip: 7092
title: Financial Bonds

> description: This interface defines a specification for financial bonds tokenization

Reviewer: This description restates the title too much. Try to add new information in the description that isn't in the title.

author: Samuel Gwlanold Edoumou (@Edoumou)
discussions-to: https://ethereum-magicians.org/t/financial-bonds/14461
status: Draft
type: Standards Track
category: ERC
created: 2023-05-28
---

## Abstract

> The proposed standard allows for the implementation of basic functionality for fixed income financial bonds with smart contracts.
> Principal bonds characteristics such as the bond isin, the issue volume, the issue date, the maturity date, the coupon rate,
> the coupon frequency, the principal, or the day count basis are defined to allow issuing bonds in the primary market (origination),
> and different transfer functions allow to buy or sell bonds in the secondary market. The standard also providses a functionality to
> allow bonds to be approved by owners in order to be spent by third party.

Reviewer: ```suggestion
This proposal introduces fixed income financial bonds. Important bonds characteristics such as the International Securities Identification Number (ISIN), the issue volume, the issue date, the maturity date, the coupon rate,
the coupon frequency, the principal, or the day count basis are defined to allow issuing bonds in the primary market (origination), and different transfer functions allow to buy or sell bonds in the secondary market. The standard also provides a functionality to allow bonds to be approved by owners in order to be spent by third party.
```

## Motivation

Fixed income instruments are one of the asset classes that is widely used by corporations and other entities to raise funds. Bonds
are considered more secured than equity since the issuer is supposed to repay the principal at maturity in addition to coupons
that are paid to investsors.

This standard interface allows fixed income instruments to be represented as on-chain tokens, so as they can be managed through wallets,
and be used by applications like decentrailized exchanges.

> 

Reviewer: Are there any existing standards for bonds on-chain? If so, you should note why this standard is better.

## Specification

The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.

**Every contract compliant with the [ERC-7092](./eip-7092.md) MUST implement the following interface**

```solidity
pragma solidity ^0.8.0;

/**
* @title ERC-7092 Financial Bonds tandard
*/
interface IERC7092 {
    /**
    *  @notice Returns the bond isin
    */
    function isin() external view returns(string memory);

    /**

>     * @notice Returs the bond name

Reviewer 2: typo "Returns"

    */
    function name() external view returns(string memory);

    /**
    * @notice Returns the bond symbol
    */
    function symbol() external view returns(string memory);

    /**

>     * @notice Returns the numbr of decimals the bond uses - e?g `10`, means to divide the token amount by `10000000000`

Reviewer 2: typo: "number of decimals"

    *
    * OPTIONAL
    */
    function decimals() external view returns(uint8);

    /**
    * @notice Returns the bond currency. This is the contract address of the token used to pay and return the bond principal
    */
    function currency() external view returns(address);

    /**
    * @notice Returns the copoun currency. This is the contract address of the token used to pay coupons. It can be same as the the one used for the principal
    */
    function currencyOfCoupon() external view returns(address);

    /**

>     * @notice Returns the bond denominiation. This is the minimum amount in which the Bonds may be issued. It must be expressend in unnit of the principal currency

Reviewer 2: typo: instead of  "unnit" it should be "uint"

Author: Yes @fosgate29 it should.

Thx

institutional issuance, the smart contracts can be managed by the investment bank. Decentralized exchanges may also use the bond standard
to list bonds, in that case, decentralized exchanges will be in charge of managing the smart contracts. Other entities may also create
tokenized bonds by integrating this financial bond interface.

Tokenizing bonds will offer several advantages compared to traditional bond issuance and trading, among with:

1. Fractional ownership: The bond standard does not limit the bond denomination to some minimum value compared to traditioanal bonds where the denomination is typically equal to $100 or $1,000.
2. Accessibility: By allowing lower investment thresholds, tokenized bonds are supposed to attract retail investors who could not participate in traditional markets due to high minimum investment requirements.
3. Increased liquidity: Fractioanal ownership will bring new investors in the bond market, this will increase liquidity in the bond market.
4. Cost savings: By replacing intermediaries with smart contracts, bond's tokenization will reduce costs associated with the bond issuance and management.

> 5. Easy accessibility and 24/7 trading: Tokenized bonds are supposed to be traded on digital platforms such as decentralized exchanges. Therefore, they will be more accessible compared to tradional bond market.

Reviewer: This list belongs in motivation. The rationale section should describe technical choices made within the EIP itself, while the motivation should justify the existence of the EIP as a whole.

A good example of what you _should_ include in your rationale section is why you don't extend an existing token standard, like ERC-20 or ERC-1155.

## Reference Implementation

The reference implementation of the [ERC-7092](./eip-7092.md) can be found [here](../assets/eip-7092/ERC7092.sol).

Some bonds have embedded options attached to them. As an example we can cite:

1. Callable bonds that have an option that gives the issuer the right to retire bonds before they mature.
2. Puttable bonds that have an option that gives investors the right to retire bonds before they mature.
3. Convertible bonds that gives investors the right to convert their bonds to equity.

Bonds with embedded options can be created by inheriting from the basic ERC-7092 that integrates the proposed interface.

> CALLABLE BONDS:

Reviewer: You can use a third level heading here (`### Callable Bonds`)

---
eip: 7092
title: Financial Bonds

> description: This interface defines a specification for debt issued by corporations, Goverments or other entities to investors in order to raise funds.

Reviewer 3: ```suggestion
description: This interface defines a specification for debt issued by corporations, governments, or other entities to investors in order to raise funds.
```

author: Samuel Gwlanold Edoumou (@Edoumou)
discussions-to: https://ethereum-magicians.org/t/financial-bonds/14461
status: Draft
type: Standards Track
category: ERC
created: 2023-05-28
---

## Abstract

> This proposal introduces fixed income financial bonds. Important bonds characteristics such as the International Securities Identification Number (ISIN),

Reviewer 3: ```suggestion
This proposal introduces fixed income financial bonds. Important bond characteristics such as the International Securities Identification Number (ISIN),
```
singular?

the issue volume, the issue date, the maturity date, the coupon rate, the coupon frequency, the principal, or the day count basis are defined to allow issuing
bonds in the primary market (origination), and different transfer functions allow to buy or sell bonds in the secondary market. The standard also provides a
functionality to allow bonds to be approved by owners in order to be spent by third party.

## Motivation

> Fixed income instruments are one of the asset classes that is widely used by corporations and other entities to raise funds. Bonds

Reviewer 3: ```suggestion
Fixed income instruments is one of the asset classes that is widely used by corporations and other entities to raise funds. Bonds
```

3. Increased liquidity: Fractioanal ownership will bring new investors in the bond market, this will increase liquidity in the bond market.
4. Cost savings: By replacing intermediaries with smart contracts, bond's tokenization will reduce costs associated with the bond issuance and management.
5. Easy accessibility and 24/7 trading: Tokenized bonds are supposed to be traded on digital platforms such as decentralized exchanges. Therefore, they will be more accessible compared to tradional bond market.

## Specification

The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.

**Every contract compliant with the [ERC-7092](./eip-7092.md) MUST implement the following interface**

```solidity
pragma solidity ^0.8.0;

/**
* @title ERC-7092 Financial Bonds tandard
*/
interface IERC7092 {
    /**
    *  @notice Returns the bond isin
    */
    function isin() external view returns(string memory);

    /**

>     * @notice Returs the bond name

Reviewer 3: ```suggestion
    * @notice Returns the bond name
```

    */
    function name() external view returns(string memory);

    /**
    * @notice Returns the bond symbol
    *         It is RECOMMENDED to represent the symbol as a combination of the issuer Issuer'shorter name and the maturity date
    *         Ex: If a company named Green Energy issues bonds that will mature on october 25, 2030, the bond symbol could be `GE30` or `GE2030` or `GE102530`
    *
    * OPTIONAL - interfaces and other contracts MUST NOT expect these values to be present. The method is used to improve usability.
    */
    function symbol() external view returns(string memory);

    /**

>     * @notice Returns the numbr of decimals the bond uses - e?g `10`, means to divide the token amount by `10000000000`

Reviewer 3: ```suggestion
    * @notice Returns the number of decimals the bond uses - e?g `10`, means to divide the token amount by `10000000000`
```

the `denomination`, and the `maturity date`. Those information are very important for bondholders to know the return on the capital invested.

### Total Supply

We made the choice not to define the `totalSupply` function beacause it can be derived from both the `issue volume` and the `denomination`.
However it is RECOMMENDED to define the `totalSupply` function in any contract that implements this standard. In that case,
the `total supply` MUST be equal to the ratio of the `issue volume` and the `denomination`.

```javascript
    totalSupply = issueVolume / denomination
```

### Account Balance

Because the balance of an account can be derived from both the `principal` and the `denomination`, we made the choice not to define the `balanceOf` function.
However it is RECOMMENDED to define the `balanceOf` function in any contract that implements this standard. In that case,
the `balance` of an account MUST be equal to the ratio of the `principal` of that account and the `denomination`.

```javascript
    balanceOf(account) = principalOf(account) / denomination
```

> ### Backwards compatibility

Reviewer 3: ```suggestion
## Backwards compatibility
```

Author: ## Backwards compatibility

doesn't pass the EIP Validator test. This is probably due to the fact that this is not defined in the template and should then be defined as a sub-sub-section.

For this reason, I will keep it as defined previously, i.e ## Backwards compatibility for the test to pass.

Reviewer 3: Ah you need to capitalize "Compatibility". Using `###` is incorrect.

    balanceOf(account) = principalOf(account) / denomination
```

> ## Backwards compatibility

Reviewer 3: must be `## Backwards Compatibility`

Author: Hello @lightclient 

When I use `## Backwards Compatibility` the test fails. The error message is 'An extra section has been added'. The backward compatibility doesn't appear in the template, that's probably the reason why when we add it as a section with ## the test fails

Author: Yes, I think there was a typo in the previous one. the previous one was `Backwards compatibility`, but it should be `Backwards Compatibility` with capital `C`.

I just corrected it.
